### DISCUSSION TITLE
Recovering gas fee for the undeployed zksync contract

### DISCUSSION THREAD

**Discussion by Hepcat93** [Cyfrin/foundry-full-course-cu#2660] [Q&A]
Hello everyone! Currently everything is fine. I worked out the process and successfully deployed my contract in Sepolia ZKSync testnet. But during my previous unsuccessful attempt I paid gas fee, although there were no traces of this contract on-chain and yet I didn't have my fee recovered. Is this normal?

Sorry guys, no screenshots, becuase I wasn't mindful enough. What I can say is that the last message in terminal was "Deploying contract", but in the left-side of the remix screen, in ZKSync's respective sub-menu there was the sign of unsucessful attept, I pointed with the mouse coursor on it, the field with tip emerged, but the text wasn't displaying at all. So buggy... I didn't expect.

**Comment by BowtiedHarpyEagle** [comment]
ZKSync stuff can still be quite buggy, therefore they are totally optional for the course. Feel free to just move along with the course, despite any issues.

**Comment by EngrPips** [comment]
Indeed

**Comment by Hepcat93** [comment]
I got you guys, thank you!

**Reply by EngrPips** [reply]
Anytime, Friend.
